### DISCUSSION TITLE
Remote Information

### DISCUSSION THREAD

**Discussion by brandonrcampbell** [wqweto/VbAsyncSocket#21] [Ideas]
This is a great replacement for winsock! Any plans to add remote information to this like resolve hostname, ip address, ping etc?

**Comment by wqweto** [comment]
`cAsyncSocket` class already (asynchronously) resolves hostnames it connects to and there is an `OnResolve` event raised.

Remote IP address and remote port can be retrieved with `GetSockName` method.

No plans for ping as ICMP protocol is completely separate from IP/UDP winsockets.
